compiler/rustc_ty/src/needs_drop.rs

```diff
@@ -21,6 +21,14 @@ fn needs_drop_raw<'tcx>(tcx: TyCtxt<'tcx>, query: ty::ParamEnvAnd<'tcx, Ty<'tcx>
     res
 }
 
+fn needs_finalizer_raw<'tcx>(tcx: TyCtxt<'tcx>, query: ty::ParamEnvAnd<'tcx, Ty<'tcx>>) -> bool {
+    let adt_fields =
+        move |adt_def: &ty::AdtDef| tcx.adt_finalize_tys(adt_def.did).map(|tys| tys.iter());
+    let res = NeedsDropTypes::new(tcx, query.param_env, query.value, adt_fields).next().is_some();
+    debug!("needs_finalize_raw({:?}) = {:?}", query, res);
```

Comment:
Is this `debug!` still needed?

Reply:
And I wasn't sure about some of the other `debug!`s too...

Reply:
Not necessarily, but they mirror the `debug!` logging that exists in the `needs_drop` impl. It is quite useful sometimes compile with the debug flag and see how rustc made its decision about whether something needed finalizing.

Reply:
OK.